### ISSUE TITLE
Generalization problem

### ISSUE THREAD

**Issue by SchweitzerGAO** [Turoad/CLRNet#17]
Description of trained model: **10 epoch** on **tusimple** dataset, accuracy **95.5%** 
I did some experiments on this trained model with my own images and images of CULane, it performed badly:
_my own image_:
![bccc0ecccc278f00c4e0e672f6df40e](https://user-images.githubusercontent.com/64410968/167835740-70240a38-6286-4aa7-bcec-e930044a2954.png)

_image in CULane_
![a6c9f125072f14efc9c54bc7650d759](https://user-images.githubusercontent.com/64410968/167836013-e589bd1c-637b-4231-b5f1-f13de417bb18.png)
I am wondering why it's performing like this. Is increasing the number of epoch useful to solve this?

**Comment by SchweitzerGAO** [comment]
Is this something like overfitting?

**Comment by Turoad** [comment]
It is underfitting.
Tusimple only has near 3000 training images and it only has highway scene.  Training using Tusimple cannot get good generalization for other complex datasets.

**Comment by SchweitzerGAO** [comment]
> It is underfitting. Tusimple only has near 3000 training images and it only has highway scene. Training using Tusimple cannot get good generalization for other complex datasets.

Thanks but still I am wondering how its performance can be improved. Is increasing the number of epochs or finetune the model with other datasets useful for this?
By the way, are there any accessible pretrained models trained by CULane dataset?

**Comment by MySuperSoul** [comment]
You can find all pretrained model in here: https://github.com/Turoad/CLRNet/releases/tag/models

**Comment by SchweitzerGAO** [comment]
> You can find all pretrained models in here: https://github.com/Turoad/CLRNet/releases/tag/models

Thanks, I will then test with the pretrained models to check if it's better

**Comment by SchweitzerGAO** [comment]
I tested with the pretrained model(DLA34 CULane), it still performed badly......
_on Tusimple_
![image](https://user-images.githubusercontent.com/64410968/168082195-5eece9b0-aaa7-4e3b-a8ad-87647b547164.png)
There's even no output on my own image:
![image](https://user-images.githubusercontent.com/64410968/168082428-4ad72300-4a3e-465b-9ad0-9a1e14a97dd7.png)
I am wondering why it is performing like this. **Is there any advice to improve the generalization performance?**

**Comment by SchweitzerGAO** [comment]
I will try this and thanks!
By the way, how did you test the image outside the dataset with `main.py`? I think it didn't provide this function. I just test this with `detect.py` in the lanedet repo.

**Comment by SchweitzerGAO** [comment]
Oh, I think I tested images from the Internet, but it performed badly. I am wondering how the performance on images outside the dataset can be improved

**Comment by SchweitzerGAO** [comment]
I tried this command:
`python main.py configs/clrnet/clr_resnet18_tusimple.py --load_from culane_r18.pth --validate --view --gpus 0`
but it failed with the error
![image](https://user-images.githubusercontent.com/64410968/168455202-947a650e-91a6-4c13-b16e-31d6e22cd6e5.png)
I wonder why it happened

**Comment by MySuperSoul** [comment]
> I tried this command: `python main.py configs/clrnet/clr_resnet18_tusimple.py --load_from culane_r18.pth --validate --view --gpus 0` but it failed with the error ![image](https://user-images.githubusercontent.com/64410968/168455202-947a650e-91a6-4c13-b16e-31d6e22cd6e5.png) I wonder why it happened

Just modify to `num_classes = 4 + 1` in config `clr_resnet18_tusimple.py`

**Comment by SchweitzerGAO** [comment]
OK, I will do this and test&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Yifei ***@***.***&gt;; 
发送时间: 2022年5月15日(星期天) 下午2:25
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Turoad/CLRNet] Generalization problem (Issue #17)





  
I tried this command: python main.py configs/clrnet/clr_resnet18_tusimple.py --load_from culane_r18.pth --validate --view --gpus 0 but it failed with the error  I wonder why it happened
  
Just modify to num_classes = 4 + 1 in config clr_resnet18_tusimple.py

**Comment by Turoad** [comment]
@SchweitzerGAO You can refer this issue https://github.com/Turoad/CLRNet/issues/19.

**Comment by SchweitzerGAO** [comment]
Thank you




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年5月15日(星期天) 晚上8:26
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Turoad/CLRNet] Generalization problem (Issue #17)





 
@SchweitzerGAO You can refer this issue #19.
